src/main/java/seedu/address/model/applicant/Applicant.java

```diff
@@ -17,16 +20,42 @@
 
     // Data fields
     private final Address address;
+    private Application application;
```

Comment:
`Applicant` is an immutable class (see class javadoc), we need to reconsider our design after v1.2

Reply:
```suggestion
    private Application application = null;
```

Can set this to null initially & remove the dummy `Application` from the constructor, since it will always be set to non-null in `ModelManager::addApplicantToPosition`.